### ISSUE TITLE
prop-types does not recognize PropTypes.exact

### ISSUE THREAD

**Issue by golopot** [yannickcr/eslint-plugin-react#2590]
`PropTypes.exact` should work just like `PropTypes.shape`. But this plugin does not recognize `PropTypes.exact`, so no error is detected.

```js
function Zoo(props) {
  return (
    <>
      {prop.foo.c} // Expects a report here.
    </>
  );
}

Zoo.propTypes = {
  foo: PropTypes.exact({
    a: PropTypes.number,
    b: PropTypes.number,
  }),
};
```

**Comment by ljharb** [comment]
That means your installed prop-types package isn’t new enough to have it.

**Comment by golopot** [comment]
I think the version of the installed prop-types package is irrelevant to this issue, I am not aware that we have version detect on prop-types package in this plugin. This plugin is able to recognize `PropTypes.Shape` because of this line https://github.com/yannickcr/eslint-plugin-react/blob/fa863ce86138b53c331731f4dd0467624206df1d/lib/util/propTypes.js#L361-L362. But there is no similar logic for `PropTypes.exact` (cannot find "exact" in that file). I think that's the problem.

**Comment by ljharb** [comment]
you’re right, PropTypes.exact should always be included in propWrapperFunctions: https://github.com/yannickcr/eslint-plugin-react/blob/3d6b371253ead2e46a8ecbb9e31b3598cfecad55/lib/util/propWrapper.js but only when it exists in the prop-types package.

**Comment by jzabala** [comment]
I opened a PR to handle the [PropTypes.exact][exact] case.

[exact]: https://reactjs.org/docs/typechecking-with-proptypes.html
